### PR TITLE
Do not require Xclang in saargs

### LINKED ISSUE #1671: do not require Xclang in saargs
it is too clumsy to give -Xclang in the saargs file before every configuration switch and value. Do not require the user to enter -Xclang.

### PR DESCRIPTION
**Pull request by csordasmarton**
> Closes #1671

Do not require the user to enter `-Xclang` before each value in saargs file.

### CODE REVIEW

File: tests/functional/analyze_and_parse/test_files/saargs_forward_multiple.cpp [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+{
+
+#if defined(HAVE_NULL_DEREFERENCE1) && defined(HAVE_NULL_DEREFERENCE2)
+  int *ptr = nullptr;
+#else
+  int x = 0;
+  int *ptr = &x;
+#endif
+
+  return *ptr;
```
[the remark this answers is not in the document]
Comment (whisperity) [reply tests/functional/analyze_and_parse/test_files/saargs_forward_multiple.cpp]: :rofl:

**Comment by whisperity** [comment]
What happens to SA-Args that don't need an `-Xclang` prefix, such as those in the documentation: https://github.com/Ericsson/codechecker/blob/master/docs/user_guide.md#-clang-static-analyzer

File: tests/functional/analyze_and_parse/test_files/saargs_forward_multiple.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (whisperity) [reply tests/functional/analyze_and_parse/test_files/saargs_forward_multiple.cpp]: *TIL*: Codacy, even if it does path-sensitive analysis, does not print the bug path or the macro stack. This is only a conditional nullptr... :slightly_frowning_face:

**Comment by Xazax-hun** [comment]
Note that the cc1 and driver flags are inherently different. There are some driver flags that makes no sense to pass to cc1 and there are flags that can be passed both to the driver and to cc1 but has slightly different meanig. Can we still pass arguments to the driver after this change?

**Comment by dkrupp** [comment]
We decided to keep the current format so, both clang and clang SA parameters can passed in saargs.

**Pull request closed without merging**